### PR TITLE
test(relocation): Improve tests

### PR DESCRIPTION
**Pull request by azaslavsky**
Some improvements:

- Add a test to ensure that paused relocations are accounted for when checking that there is already a relocation in flight, fixing a bug.
- Clean up test names a bit.
- List JSONField instances were being passed as strings; these are not lists in all test cases.

### CODE REVIEW

File: tests/sentry/api/endpoints/relocations/test_index.py [hunk 1]
```diff
@@ -495,7 +496,7 @@ def test_fail_expired_superuser_when_feature_disabled(self):
     ]:
 
         @patch("sentry.tasks.relocation.uploading_complete.delay")
-        def test_success_good_org_slugs(
+        def test_good_good_org_slugs(
```
Comment (hubertdeng123) [inline tests/sentry/api/endpoints/relocations/test_index.py:499]:
```suggestion
        def test_good_valid_org_slugs(
```

File: tests/sentry/api/endpoints/relocations/test_index.py [hunk 2]
```diff
@@ -539,7 +540,7 @@ def test_success_good_org_slugs(
     ]:
 
         @patch("sentry.tasks.relocation.uploading_complete.delay")
-        def test_fail_bad_org_slugs(
+        def test_bad_bad_org_slugs(
```
Comment (hubertdeng123) [inline tests/sentry/api/endpoints/relocations/test_index.py:543]:
```suggestion
        def test_bad_invalid_org_slugs(
```

File: tests/sentry/api/endpoints/relocations/test_abort.py [hunk 3]
```diff
@@ -28,8 +28,8 @@ def setUp(self):
             owner_id=self.owner.id,
             status=Relocation.Status.IN_PROGRESS.value,
             step=Relocation.Step.PREPROCESSING.value,
-            want_org_slugs='["foo"]',
```
Comment (hubertdeng123) [inline tests/sentry/api/endpoints/relocations/test_abort.py:31]: Hmmm, should these have ever been strings? Not seeing changes reflected for this

Comment (azaslavsky) [reply tests/sentry/api/endpoints/relocations/test_abort.py:31]: I think most of these cases don't actually use this bad data, it was just there for completeness.

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**